components/suggest/src/suggestion.rs

```diff
@@ -97,8 +97,14 @@ pub enum Suggestion {
         // and therefore the only one we'll collect metrics for.
         match_info: Option<FtsMatchInfo>,
     },
-    Exposure {
+    Dynamic {
         suggestion_type: String,
+        data: Option<serde_json::Value>,
```

Comment:
Instead of storing a `Value` here, could this store an enum so that consumers don't need to deserialize the raw JSON?  That means that adding a new dynamic suggestion type would require app-services changes, but it would just be adding a new variant to the enum.

I think that would mean using `#[serde(tag = "suggestion_type")]` on the enum rather than storing the `suggestion_type` as a separate field.  I don't think there would need to that many other changes though.

Reply:
I'm also confused at how UniFFI is letting you use `serde_json::Value` like that.  Normally there needs to be some sort of custom type definition.